ambry-store/src/main/java/com/github/ambry/store/RemoteTokenTracker.java

```diff
@@ -69,7 +101,204 @@ void refreshPeerReplicaTokens() {
   /**
    * @return a snapshot of peer replica to token map.
    */
-  Map<String, FindToken> getPeerReplicaAndToken() {
+  Map<String, Pair<Long, FindToken>> getPeerReplicaAndToken() {
     return new HashMap<>(peerReplicaAndToken);
   }
+
+  /**
+   * @return the file path which persists the remote token.
+   */
+  protected String getPersistFilePath() {
+    return localReplica.getReplicaPath() + File.separator + REMOTE_TOKEN_FILE_NAME;
+  }
+
+  /**
+   * Provide a way to explicitly persist the token besides the background persistor
+   */
+  protected boolean persistToken() {
+    try {
+      persistor.persist();
+      return true;
+    } catch (IOException e) {
+      return false;
+    }
+  }
+
+  /**
+   * Disable this {@link RemoteTokenTracker} and cancel the scheduled persistor.
+   */
+  @Override
+  public void close() {
+    persistor.cancel();
+  }
```

Comment:
we can make schedule to return a future and use future to stop the execution. And then we should call persist() again to make sure we dump the latest set of tokens.

Reply:
Thanks for the suggestion. Changed.